### PR TITLE
[Docs] Re-arrange data store content

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/store/datastore.md [hunk 1]
```diff
@@ -3,27 +3,32 @@
...
 * **store** &mdash; MLRun versioned artifacts [(see Artifacts)](./artifacts.html), format: `store://artifacts/<project>/<artifact-name>[:tag]`
 * **memory** &mdash; in memory data registry for passing data within the same process, format `memory://key`, use `mlrun.datastore.set_in_memory_item(key, value)` 
    to register in memory data items (byte buffers or DataFrames). (Not supported by all Spark runtimes)
 
+**In this section**
+- [Storage credentials and parameters](#storage-credentials-and-parameters)
+- [Using data store profiles](#using-data-store-profiles)
+- [Azure](#azure)
+- [Databricks file system](#databricks-file-system)
+- [HDFS data store profile](#hdfs-data-store-profile)
```
Comment (assaf758) [inline docs/store/datastore.md]: these "TOC" titles should point to the datastore, not the datastore profile

File: docs/store/datastore.md [hunk 2]
```diff
@@ -71,70 +76,8 @@ db.create_project_secrets(project=project_name, provider="kubernetes", secrets=s
...
-  ARN must be of the format `arn:aws:iam::<account-of-role-to-assume>:role/<name-of-role>`
-* `AWS_PROFILE` &mdash; name of credentials profile from a local AWS credentials file. 
-  When using a profile, the authentication secrets (if defined) are ignored, and credentials are retrieved from the 
-  file. This option should be used for local development where AWS credentials already exist (created by `aws` CLI, for
-  example)
   
-## Using data store profiles
+## Data store profiles
 
 ```{admonition} Notes
```
Comment (assaf758) [inline docs/store/datastore.md:80]: in my opinion a note about lack of support in specific use-case should be at then end of chapter, when datastore profile is already explained

File: docs/store/datastore.md [hunk 3]
```diff
@@ -228,8 +205,10 @@ The equivalent to this parameter in environment authentication is "DATABRICKS_HO
 For privacy reasons, it's tagged as a private attribute, and its default value is `None`.
 The equivalent to this parameter in environment authentication is "DATABRICKS_TOKEN".
 
+## Google cloud storage
 ### GCS data store profile
```
Comment (assaf758) [inline docs/store/datastore.md:217]: Here you are describing the profile before the parameters.
I suggest you keep it consistent - first the parameters, then the profile.

File: docs/store/datastore.md [hunk 4]
```diff
@@ -244,31 +223,52 @@ The equivalent to this parameter in environment authentication is "GCP_CREDENTIA
...
+### GCS credentials and parameters
+* `GOOGLE_APPLICATION_CREDENTIALS` &mdash; path to the application credentials to use (in the form of a JSON file). This can
+be used if this file is located in a location on shared storage, accessible to pods executing MLRun jobs.
+* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this secret or environment variable
+may contain the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
+and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS`'s
+contents directly to the query engine.
 
-### Kafka data store profile
+## HDFS data store profile
```
Comment (assaf758) [inline docs/store/datastore.md]: section ## HDFS data store
Here you can mention that we don't support HDFS with env-vars, only with datastore profile
subsection ### HDFS datastore profile

File: docs/store/datastore.md [hunk 5]
```diff
@@ -168,13 +111,37 @@ To access a profile from the client/sdk, register the profile locally by calling
...
 local_redis_profile = DatastoreProfileRedis(redis_profile.name, redis_profile.endpoint_url, username="mylocaluser", password="mylocalpassword")
 register_temporary_client_datastore_profile(local_redis_profile)
 ```
-### Azure data store profile
+
+
+
+
+
+## Azure
```
Comment (assaf758) [inline docs/store/datastore.md]: Azure datastore

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: docs/store/datastore.md (hunk 2 above)
Comment (jillnogold) [reply docs/store/datastore.md:80]: my theory was that you don't want to read all this if it's not relevant for you. but moving it to the bottom

**Review by assaf758 — APPROVED** [review]

**Pull request merged**